### PR TITLE
Refactor code to use the REST API rather than GraphQL

### PR DESCRIPTION
**Pull request by egibs**
We've hit a point where using the undocumented, unsupported GraphQL API is causing more issues than it solves, even though it offers more information.

This PR moves over to using the documented (albeit, under-baked) REST API and reworks how noncompliant users are retrieved.

The filtered users are returned as a map with the unique ID as a key and the corresponding attributes as a nested map. These attributes are then used when notifying users.

Caveats:
- Handling due dates is TBD (there aren't any dates around training completions or the like)
- The API is missing a way to dynamically retrieve the Company name (hence the move to a static string flag)
- The authentication header requires both the access key and secret key separated by a space (and only Secureframe Super Admins can see the access key when creating an API key; Admins can see the secret key, oddly enough)

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -163,40 +159,36 @@ func main() {
...
-		if !requiredTypes[eType] {
-			continue
-		}
+		// Instructions to be populated based on the user's onboarding status
+		var needs []string
 
-		needs := []string{}
-		if !p.PoliciesAccepted {
+		// Account for users with training that has not started or with outstanding security training
+		switch info["onboarding_status"] {
```
Comment (egibs) [inline main.go:161]: I'm not sure what the possible values for this are. When testing, I only saw `all_tasks_completed`, `not_started`, and `security_training`.

File: pkg/secureframe/api.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	users := make(map[string]map[string]string)
+
+	requestUrl := fmt.Sprintf("%s/users", restEndpoint)
+
+	rb, err := makeRequest(ctx, requestUrl, "GET", accessKey, secretKey)
+	if err != nil {
+		return nil, err
+	}
+
+	// Parse response bytes using Gabs to avoid defining structs
```
Comment (tstromberg) [inline pkg/secureframe/api.go]: What do you think about using https://mholt.github.io/json-to-go/ to generate a Go struct from the JSON data? That would eliminate all the awkward data copying this code is doing to generate vars and then a map of strings - as well as eliminate an unusual code dependency.

File: pkg/secureframe/api.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	rb, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	return rb, nil
+}
+
+// GetUsers returns a map of noncompliant users and related information.
+func GetUsers(ctx context.Context, accessKey, secretKey, types string) (map[string]map[string]string, error) {
```
Comment (tstromberg) [inline pkg/secureframe/api.go]: Avoid the prefix "Get":

https://google.github.io/styleguide/go/best-practices.html#naming-conventions

The genericization of this rule for go is: if you can name a function a Verb, do so. If you can't, give it a Noun. Give it both when necessary for clarity.

**Review by tstromberg — CHANGES_REQUESTED** [review]

File: pkg/secureframe/api.go (hunk 2 above)
Comment (egibs) [reply pkg/secureframe/api.go]: Updated in [`43b421d` (#4)](https://github.com/chainguard-dev/secureframe-policy-minder/pull/4/commits/43b421d147a39786d38ea0b32986443daa885a12).

I've used Gabs in the past but I agree that it's not great with all of the type assertions. The new way is definitely cleaner.

File: pkg/secureframe/api.go (hunk 3 above)
Comment (egibs) [reply pkg/secureframe/api.go]: Updated in [`43b421d` (#4)](https://github.com/chainguard-dev/secureframe-policy-minder/pull/4/commits/43b421d147a39786d38ea0b32986443daa885a12).

**Review by tstromberg — APPROVED** [review]

**Pull request merged**